feeder/ethereum/src/getUpdateDataForLog.ts

```diff
@@ -0,0 +1,40 @@
+import LogClient from "./LogClient";
+import RootWitnessClient from "./RootWitnessClient";
+import { Log, UpdateData, hashString } from "./utils";
+
+export async function getUpdateDataForLog(
+  log: Log,
+  rootWitnessClient: RootWitnessClient
+): Promise<UpdateData | undefined> {
+  console.log("Processing log\n", log);
+  const { id, url } = log;
+  const [
+    { size: witnessedSize, timestamp: witnessedTimestamp },
+    { hash, signature, treeSize: currentSize, timestamp },
+  ] = await Promise.all([
+    rootWitnessClient.getTreeDataById(id),
+    LogClient.getSth(url),
+  ]);
+  console.log(`Found current state of log ${id}`, {
+    witnessedSize: witnessedSize.toNumber(),
+    witnessedTimestamp: witnessedTimestamp.toNumber(),
+    hash,
+    signature,
+    currentSize,
+    timestamp,
+    id,
+  });
+  if (timestamp.toString() == witnessedTimestamp.toString()) {
```

Comment:
What should happen if the new STH timestamp is smaller than the witnessed one (e.g. because the log frontend is "stuck" or we're behind an aggressive cache?

Reply:
Good catch; I've updated it to bailout in that scenario as well.

FYI I moved this function into `Feeder.ts`